Firestore/core/src/remote/bloom_filter.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2023 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIREBASE_CORE_SRC_REMOTE_BLOOM_FILTER_H
+#define FIREBASE_CORE_SRC_REMOTE_BLOOM_FILTER_H
+
+#include <string>
+#include <vector>
+#include "absl/strings/string_view.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+class BloomFilter final {
+ public:
+  BloomFilter(std::vector<uint8_t> bitmap, int32_t padding, int32_t hash_count);
```

Comment:
should I set the padding and hash_count type to uint_32, to get rid of the negative value validation in constructor implementation?